File: examples/java/Tree1.java
public class Tree1 {
    static class Node {
        int data;
        Node left, right;
        Node(int d) { data = d; left = right = null; }
    }

    static Node root;

    static Node insert(Node n, int val) {
        if (n == null) return new Node(val);
        if (val < n.data) n.left = insert(n.left, val);
        else n.right = insert(n.right, val);
        return n;
    }

    static void inorder(Node n) {
        if (n == null) return;
        inorder(n.left);
        System.out.print(n.data + " ");
        inorder(n.right);
    }

    public static void main(String[] args) {
        root = insert(root, 50);
        insert(root, 30);
        insert(root, 70);
        insert(root, 20);
        insert(root, 40);
        inorder(root);
    }
}

File: examples/csharp/Tree1.cs
using System;

class Tree1
{
    class Node { public int Data; public Node Left, Right; }

    static void Main()
    {
        Node root = null;
        root = Insert(root, 5);
        root = Insert(root, 3);
        root = Insert(root, 7);
        root = Insert(root, 1);
        Inorder(root);
        Console.WriteLine();
    }

    static Node Insert(Node node, int data)
    {
        if (node == null) return new Node { Data = data };
        if (data < node.Data) node.Left = Insert(node.Left, data);
        else node.Right = Insert(node.Right, data);
        return node;
    }

    static void Inorder(Node node)
    {
        if (node == null) return;
        Inorder(node.Left);
        Console.Write(node.Data + " ");
        Inorder(node.Right);
    }
}
